### PR TITLE
Adopt the Xanadu Sphinx Theme

### PR DESCRIPTION
**Pull request by Mandrenkov**
**Context:**
Currently, all Xanadu OSS projects include their own version of the Xanadu Sphinx theme; however, the [Xanadu Sphinx Theme](https://github.com/XanaduAI/xanadu-sphinx-theme) repository is now publicly available and is the preferred way to access the Xanadu CSS theme and Sphinx directives.

**Description of the Change:**
* Deleted the doc/xanadu_theme directory.
* Updated doc/requirements.txt and doc/conf.py to reference and use the (centralized) Xanadu Sphinx Theme.

**Benefits:**
* The size of this repository is smaller, meaning `git clone` will run faster.
* Updates to the Xanadu Sphinx theme no longer have to be manually ported to this repository.

**Possible Drawbacks:**
None.

**Related GitHub Issues:**
Depends on XanaduAI/xanadu-sphinx-theme#2.

**Note to Reviewers:**
The _Install_ and _Circuits_ pages capture most of the representative changes between the Sphinx themes.

### CODE REVIEW

**Comment by josh146** [comment]
❤️

A couple of things I noticed while browsing around:

- The logo and wordmark in the top left corner (image) should link to https://strawberryfields.ai. This is the first time I noticed this, since this is the first repo where the homepage != the docs homepage.

- In the `master` branch, autosummary tables are [fullwidth](https://strawberryfields.readthedocs.io/en/stable/code/sf.html), but this is not the case in the [centralized theme](https://strawberryfields--701.org.readthedocs.build/en/701/code/sf.html).

**Review by thisac — APPROVED** [review]
Just a minor thing. I noticed the hover text isn't visible when on the edge (not sure how easy this would be to fix).
Screen Shot 2022-04-14 at 11 05 50 AM
 
Also, a bummer that the font in the top left isn't "the Xanadu font" which was visible before when using an image instead.

**Comment by Mandrenkov** [comment]
Thanks for the review, @josh146!

> * The logo and wordmark in the top left corner should link to https://strawberryfields.ai. This is the first time I noticed this, since this is the first repo where the homepage != the docs homepage.

Nice catch! 🎣 I've updated the theme to support custom `href` targets for the logo and wordmark in the navigation bar.

> * In the `master` branch, autosummary tables are [fullwidth](https://strawberryfields.readthedocs.io/en/stable/code/sf.html), but this is not the case in the [centralized theme](https://strawberryfields--701.org.readthedocs.build/en/701/code/sf.html).

Another great observation! 🔎 This has been fixed in the latest version of the centralized Xanadu Sphinx Theme.

File: doc/requirements.txt [hunk 1]
```diff
@@ -18,3 +19,5 @@ tensorflow==2.6.3
 thewalrus==0.18.0
 toml==0.10.2
 mistune==0.8.4
+# TODO: Use the Xanadu-Sphinx-Theme package on PyPI once it has been released.
+git+https://github.com/XanaduAI/xanadu-sphinx-theme@phoenix#egg=xanadu_sphinx_theme
```
[the remark this answers is not in the document]
Comment (Mandrenkov) [reply doc/requirements.txt:23]: Yep. Fixed!

**Comment by Mandrenkov** [comment]
Thanks, @thisac!

> Just a minor thing. I noticed the hover text isn't visible when on the edge (not sure how easy this would be to fix).

Good eye! 👁 I don't think this would be too difficult to fix; however, I would prefer to just remove the tooltips as they (mostly) just repeat the title of the gallery item anyway.

> Also, a bummer that the font in the top left isn't "the Xanadu font" which was visible before when using an image instead.

Yes, agreed, although I stand by the decision to use a consistent font across all of our OSS projects to ensure the Sphinx assets look crisp and unified. If only we had access to the Xanadu wordmark font...

**Comment by Mandrenkov** [comment]
As an update @thisac, after speaking with @josh146, I've restored the SF wordmark in the navigation bar.

**Comment by josh146** [comment]
Thanks @Mandrenkov!

**Pull request merged**